src/Evolution/DgSubcell/Matrices.cpp

```diff
@@ -283,9 +283,10 @@ Matrix reconstruction_matrix_cache_impl_helper(
   // than Blaze.
```

Comment:
>   // Do matrix multipy with dgemm_ directly because that seems to be faster
>   // than Blaze

@nilsdeppe The reason for this could be that we hadn't enabled BLAS kernels for Blaze. You could check again now that it's enabled.

Reply:
Isn't Blaze hard-coded to use cblas? I'm not sure how the header choice effects the implementation, but cblas is pretty much always the slowest blas implementation possible. In general we'll want to use MKL or OpenBLAS.

Reply:
Blaze uses whichever BLAS you link in. The header doesn't affect this, it only defines the interface (standardized by BLAS). That's why many people don't even bother including the header but call the BLAS functions "blindly". The Blaze docs say you should enable BLAS for max performance; in particular the BLAS dense matrix multiplication kernel is faster than Blaze's fallback: https://bitbucket.org/blaze-lib/blaze/wiki/Configuration%20and%20Installation#!step-2-configuration

That could be the reason why you found Blaze performed worse than BLAS for this particular case. With BLAS enabled in Blaze you should be able to use matrix multiplication directly here.